app/src/main/java/comcmput301f16t01/github/carrier/DriverViewRequestActivity.java

```diff
@@ -1,21 +1,282 @@
+                    minLength = road.mLength;
+                    bestRoad = road;
+                }
+            }
+
+            String routeDesc = bestRoad.getLengthDurationText(activity, -1);
+            Polyline roadPolyline = RoadManager.buildRoadOverlay(bestRoad);
+            roadPolyline.setTitle(getString(R.string.app_name) + " - " + routeDesc);
+            roadPolyline.setInfoWindow(new BasicInfoWindow(org.osmdroid.bonuspack.R.layout.bonuspack_bubble, map));
+            mapOverlays.add(0, roadPolyline);
+            map.invalidate();
+        }
+    }
+
+    /**
+     * Given the request passed in by the user, set the views in the layout.
+     * @param request
+     */
+    public void setViews(Request request) {
+        UserController uc = new UserController();
+
+        // Set up the fare
+        FareCalculator fc = new FareCalculator();
+        TextView fareTextView = (TextView) findViewById(R.id.textView_$fareAmount);
+        fareTextView.setText("$" + fc.toString(request.getFare()));
+
+        // Set up the UsernameTextView of the rider
+        UsernameTextView riderUsernameTextView = (UsernameTextView) findViewById(R.id.UsernameTextView_rider);
+        riderUsernameTextView.setText(request.getRider().getUsername());
+        riderUsernameTextView.setUser(request.getRider());
+
+        // Set up the UsernameTextView of the driver
+        UsernameTextView driverUsernameTextView = (UsernameTextView) findViewById(R.id.UsernameTextView_driver);
+        driverUsernameTextView.setText(uc.getLoggedInUser().getUsername());
+        driverUsernameTextView.setUser(uc.getLoggedInUser());
+
+        TextView startAddressTextView = (TextView) findViewById(R.id.textView_start);
+        String startAddress = request.getStart().getAddress();
+        if(startAddress != null) {
+            startAddressTextView.setText(startAddress);
+        } else {
+            String startPoint = "(" + String.valueOf(request.getStart().getLatitude()) + ", " +
+                    String.valueOf(request.getStart().getLongitude()) + ")";
+            startAddressTextView.setText(startPoint);
+        }
+
+        TextView endAddressTextView = (TextView) findViewById(R.id.textView_end);
+        String endAddress = request.getEnd().getAddress();
+        if(endAddress != null) {
+            endAddressTextView.setText(request.getEnd().getAddress());
+        } else {
+            String endPoint = "(" + String.valueOf(request.getEnd().getLatitude()) + ", " +
+                    String.valueOf(request.getEnd().getLongitude()) + ")";
+            endAddressTextView.setText(endPoint);
+        }
+
+        TextView descriptionTextView = (TextView) findViewById(R.id.textView_description);
+        descriptionTextView.setText(request.getDescription());
+
+        /**
+         * This switch statement changes the status image
+         * There is currently coloring code in here which we may use
+         * in the future for a notification system?
```

Comment:
Could you do me a favour? We do not have to dio the color things anymore. Would you either delete it or put a TODO to delete color stuff?